eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/config/CcConfiguration.java

```diff
@@ -10,305 +17,394 @@
 import org.eclipse.core.resources.ProjectScope;
 import org.eclipse.core.resources.ResourcesPlugin;
 import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.preferences.ConfigurationScope;
 import org.eclipse.core.runtime.preferences.IEclipsePreferences;
 import org.eclipse.core.runtime.preferences.IScopeContext;
 import org.osgi.service.prefs.BackingStoreException;
 
+import com.google.common.collect.Sets;
+
+import cc.codechecker.plugin.CodeCheckerNature;
+import cc.codechecker.plugin.Logger;
 import cc.codechecker.plugin.config.Config.ConfigTypes;
+import cc.codechecker.plugin.config.Config.ConfigTypesCommon;
+import cc.codechecker.plugin.config.Config.ConfigTypesProject;
 import cc.codechecker.plugin.runtime.CodeCheckEnvironmentChecker;
 import cc.codechecker.plugin.runtime.CodecheckerServerThread;
-import cc.codechecker.plugin.CodeCheckerNature;
-import cc.codechecker.plugin.config.CodeCheckerContext;
-import cc.codechecker.plugin.views.console.ConsoleFactory;
-
-import java.io.File;
-import java.util.HashMap;
-import java.util.Map;
-
-import cc.codechecker.plugin.Logger;
-import org.eclipse.core.runtime.IStatus;
 
+/**
+ * Stores and manages configurations related to projects and the plug-in.
+ */
 public class CcConfiguration {
 
-    //Logger
-    IProject project=null;
-    IEclipsePreferences projectPreferences=null;
-    IEclipsePreferences globalPreferences = null;
-
-    private  Map<ConfigTypes,String> defaults;//default config values
-    private  Map<ConfigTypes,String> configKeys;//textual values of config variables
+    private static Map<ConfigTypes, String> globalConfig;
+    private static IEclipsePreferences globalPreferences;
+    
+    private static final String STR_EMPTY = "";
+    private static final String TRUE = "";
+    private static final String SEP = ":";
+    
+    private static final String CODECHECKER_WORKSPACE = ResourcesPlugin.getWorkspace().getRoot().getLocation().toString() 
+            + "/.codechecker";
```

Comment:
Is there some platform independent path join in Java?

Reply:
I'll replace them with java.nio.files.* classes, it's available from Java 1.7 so I think we're good to go.